segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/AbstractTaskQueue.java

```diff
@@ -0,0 +1,41 @@
+/**
+ * Copyright Pravega Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.pravega.segmentstore.storage.chunklayer;
+
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Represents background task queue.
+ *
+ * @param <TaskType>
+ */
+public interface AbstractTaskQueue<TaskType> extends AutoCloseable {
```

Comment:
This is not a queue, but a collection of queues. 

I am trying to understand why do we need yet another abstraction for this. Could we not just have a collection of queues (index them by name), and get them by name; then add the task to those objects directly?

This interface, as it is, doesn't explain what happens if we try to add a task to an inexistent queue, for example. To be more concise, this interface (if you decide to keep it), needs a bit more documentation on what each method is doing and what is the caller's expectation on each of those futures returned by it.

Reply:
The main reason is that storage can not have direct dependency on code in container.
In addition this abstraction is at the GC concept level without need to also worry about details of serialization/deserialization etc. etc.
The third and also equally important reason is that it makes GC implementation very loosely coupled to actual underlying event processor implementation.